### PR TITLE
Add button to add as admin in project join req

### PR DESCRIPTION
**Pull request by AdityaKhatri**
Need to merge https://github.com/the-deep/server/pull/47 in server.

### CODE REVIEW

File: src/views/Notifications/ProjectJoin/index.js [hunk 1]
```diff
@@ -138,10 +139,17 @@ export default class ProjectJoinItem extends React.PureComponent {
                     <SuccessButton
                         className={styles.button}
                         iconName={iconNames.check}
-                        onClick={() => this.handleRequestApproval(true)}
+                        onClick={() => this.handleRequestApproval(true, 'normal')}
```
Comment (tnagorra) [inline src/views/Notifications/ProjectJoin/index.js]: No inline function

File: src/views/Notifications/ProjectJoin/index.js [hunk 2]
```diff
@@ -138,10 +139,17 @@ export default class ProjectJoinItem extends React.PureComponent {
...
                     >
                         {_ts('notifications.projectJoin', 'acceptButton')}
                     </SuccessButton>
+                    <SuccessButton
+                        className={styles.button}
+                        iconName={iconNames.check}
+                        onClick={() => this.handleRequestApproval(true, 'admin')}
```
Comment (tnagorra) [inline src/views/Notifications/ProjectJoin/index.js]: No inline function

File: src/views/Notifications/requests/ProjectJoinResponseRequest.js [hunk 3]
```diff
@@ -15,7 +15,8 @@ export default class ProjectJoinResponseRequest {
     }
 
     success = (response) => {
-        this.props.setProjectJoinStatus({ newNotificationDetails: response });
+        const newNotificationDetails = { ...response };
```
Comment (tnagorra) [inline src/views/Notifications/requests/ProjectJoinResponseRequest.js]: Unnecessary clone of object.

File: src/views/Notifications/ProjectJoin/index.js [hunk 4]
```diff
@@ -54,7 +54,7 @@ export default class ProjectJoinItem extends React.PureComponent {
         }
     }
 
-    startProjectJoinResponseRequest = (projectId, requestId, approval) => {
+    startProjectJoinResponseRequest = (projectId, requestId, approval, role) => {
         let response = 'reject';
```
Comment (tnagorra) [inline src/views/Notifications/ProjectJoin/index.js]: This should use ternary operator

File: src/views/Notifications/requests/ProjectJoinResponseRequest.js [hunk 5]
```diff
@@ -37,10 +37,14 @@ export default class ProjectJoinResponseRequest {
         });
     }
 
-    create = (projectId, requestId, response) => {
+    create = (projectId, requestId, response, role) => {
+        const body = {};
```
Comment (tnagorra) [inline src/views/Notifications/requests/ProjectJoinResponseRequest.js]: This should use ternary operator.

File: src/redux/initial-state/dev-lang.js [hunk 6]
```diff
@@ -524,8 +524,10 @@ const initialDevLangState = {
         1933: '{requestedBy}\'s request to join the project {project} was rejected by {respondedBy}.',
         1934: 'Project Join Response',
         1935: 'Project Join Failure',
-        1936: 'Accept',
+        1936: 'Add',
```
Comment (tnagorra) [inline src/redux/initial-state/dev-lang.js]: Delete this and use 'Add' at 301.

File: src/views/Notifications/ProjectJoin/index.js [hunk 7]
```diff
@@ -154,13 +179,17 @@ export default class ProjectJoinItem extends React.PureComponent {
             requestedBy,
             respondedBy,
             status,
+            role,
         } = data.details;
 
-        const approvalText = status === 'rejected' ? 'rejectedText' : 'acceptedText';
+        let approvalText = 'rejectedText';
+        if (status === 'accepted') {
```
Comment (tnagorra) [inline src/views/Notifications/ProjectJoin/index.js:186]: Never use variable to get string identifier. Use two _ts methods

Comment (AdityaKhatri) [reply src/views/Notifications/ProjectJoin/index.js:186]: Fixed

**Pull request merged**